### ISSUE TITLE
bug: rq_rand not working in redirect rule

### ISSUE THREAD

**Issue by jpaddock** [requestly/requestly#1877]
### Describe your issue?

I'm trying to utilize the rq_rand function in a redirect rule, but the literal string is being passed to the browser instead of the processed value. My redirect to section is `$1/dev/script.js?_cb=rq_rand(10)`.

### Repro steps

1. Create a new redirect rule.
2. URL matches Regex
3. For the redirect, include rq_rand(10) and the literal string is passed instead of the processed value.

### What Requestly tool were you using?

- [X] Extension
- [ ] Desktop-App
- [ ] Android SDK
- [ ] Selenium

### Your Environment

Chrome Version 126.0.6478.127 (Official Build) (arm64)

### Requestly Version

Latest Chrome extension

**Comment by wrongsahil** [comment]
Hey. Due to chrome extension MV3 changes, we had to deprecated the `rq_rand()` and other predefined functions. Right now i don't see a direct way of implementing this. But maybe in future depending upon the how the chrome extension API adds support for it (I'm highly doubtful though)

You can read more about the release [here](https://github.com/requestly/requestly/issues/1690).

**Comment by wrongsahil** [comment]
Good point. I don't think we removed it from Requestly Desktop App. Let me check

**Comment by wrongsahil** [comment]
Looks like we only supported `rq_rand` for header rule only in desktop 😓. Let me think of any other way if this is possible to achieve. We should be able to achieve this in desktop app though 
CC: @nsrCodes 

https://github.com/requestly/requestly/blob/3f58919e6c92252916899c6cb27a7f0b0f9260d4/common/rule-processor/src/processors/HeadersRuleProcessor.js#L41

**Comment by wrongsahil** [comment]
Hey @cuotos . This is an old blog which got recently updated. That's why the date shows Feb25. 

We'll mark the feature as deprecated for now. Thanks for pointing out.

**Comment by Ronakkadhi** [comment]
Closing as stale — no activity for over a year.

As part of the 2026 repository restructure, we're closing dormant issues to focus on actively-tracked items. If this is still relevant, please re-open or file a fresh issue at the right home:

- 🌐 **HTTP Interceptor** (Chrome extension + web app + desktop proxy) → [requestly/interceptor](https://github.com/requestly/interceptor/issues/new/choose)
- 🚀 **Requestly API Client** → this repo

Thanks for taking the time to file the original report.
